fg/settings/config.py

```diff
@@ -7,18 +7,45 @@
 with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 '''
+import os
+from scif.logger import bot
+
+def getenv(variable_key, default=None, required=False, silent=True, boolean=False):
+    ''' getenv will attempt to get an environmental variable. If the variable
+        is not found, the deafult value will be returned.
+
+        Parameters
+        ==========
+        variable_key: the variable name to be searched for
+        required: exit with error if not found
+        silent: if silent, does not print debugging information
+
+        All credit to https://github.com/vsoch/scif/blob/master/scif/defaults.py
+    '''
+    variable = os.environ.get(variable_key)
+    if variable is None and required == True:
+        bot.exit("Cannot find environment variable {}, exiting.".format(variable_key))
+
+    if silent != True and variable is not None:
+        bot.verbose("{} found as {}".format(variable_key,variable))
+
+    if boolean == True:
+        variable = variable == 'True' # Handles booleans
+
```

Comment:
This is very likely to run us into trouble, when people don't read instructions carefully (e.g., you must use "True" or "False" we are undoubtedly going to see all of the following for true:

 - true
 - 1
 - True
 - yes

So - although it's not as elegant as a one liner, we do need a function that will handle this gracefully, e.g.,

```python
def convert2boolean(arg):
    '''
    convert2boolean is used for environmental variables
    that must be returned as boolean
    '''
    if not isinstance(arg, bool):
        return arg.lower() in ("yes", "true", "t", "1", "y")
    return arg
```

and then the usage would be to remove the boolean argument from the getenv function, and just wrap the entire thing with convert2boolean (and feel free to change the name to something else more elegant).

```python
convert2boolean(getenv('SCIF_ALLOW_APPEND', True))
```

or if you like, you can have that snippet integrated into the function above:

```python
def getenv(variable_key, default=None, required=False, silent=True, boolean=False):
    ''' getenv will attempt to get an environmental variable. If the variable
        is not found, the deafult value will be returned.
        Parameters
        ==========
        variable_key: the variable name to be searched for
        required: exit with error if not found
        silent: if silent, does not print debugging information
        All credit to https://github.com/vsoch/scif/blob/master/scif/defaults.py
    '''
    variable = os.environ.get(variable_key)
    if variable is None and required:
        bot.exit("Cannot find environment variable {}, exiting.".format(variable_key))

    if boolean and variable is not None:
        if not isinstance(variable, bool):
            variable = variable.lower() in ("yes", "true", "t", "1", "y")

    if not silent and variable is not None:
        bot.verbose("{} found as {}".format(variable_key, variable))

    return variable
```
Note that I also switched the bot.verbose command so the developed can see the final parsed value (if boolean is given).

Reply:
Seemed easiest to use integrated snippet, so replaced current function with that. Question: is there a reason why you prefer `if not silent` to `if silent != True`? It seems to me like the latter is clearer on what value is expected

Reply:
Actually, my preference is for `if silent != True` but when I started using pylint, it would get mad at me for doing that. Here is an example:

```python
#!/usr/bin/env python3

'''
This is an example of using not have_avocados
'''

def main():
    '''the main function'''
    have_avocados = False

    if not have_avocados:
        print("So sad")

if __name__ == "__main__":
    main()
```
Now we run pylint with defaults:

```bash
$ pylint avocados1.py

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)
```
Now let's change the if statement:

```python
#!/usr/bin/env python3

'''
This is an example of using not have_avocados
'''

def main():
    '''the main function'''
    have_avocados = False

    if have_avocados != True:
        print("So sad")

if __name__ == "__main__":
    main()
```
And now pylint is telling us pep8 isn't happy

```python
************* Module avocados2
avocados2.py:11:7: C0121: Comparison to True should be just 'not expr' (singleton-comparison)

------------------------------------------------------------------
Your code has been rated at 8.33/10 (previous run: 8.33/10, +0.00)
```

So yeah, this is definitely one of those cases where my personal coding style / preference went against what pep8 told me I should do - and I don't do this for all the niche rules, but for this one I try to conform because it does produce functionally equivalent, cleaner code. And I've noticed other developers harping on me about it too, so I think the general consensus is to follow the simpler syntax.

Reply:
In this case if reading the line like a sentence didn't sound right, you could flip it to be something like:

```python
if not verbose:
   print("BLA bla")
```
which reads a little better than "not silent"